packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1149,6 +1160,33 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     this.addEnvironment('AWS_LAMBDA_EXEC_WRAPPER', props.adotInstrumentation.execWrapper);
   }
 
+  /**
+   * Add a Parameters and Secrets Extension Lambda layer.
+   */
+  private configureParamsAndSecretsExtension(props: FunctionProps): void {
+    if (props.paramsAndSecrets === undefined) {
+      return;
+    }
+
+    // grant permissions to lambda execution role to allow access to provided secrets
+    props.paramsAndSecrets.secrets?.forEach(secret => {
+      secret.grantRead(this);
+      if (secret.encryptionKey) {
+        secret.encryptionKey.grantDecrypt(this);
+      }
+    });
+
+    // grant permission to lambda execution role to allow access to provided parameters
+    props.paramsAndSecrets.parameters?.forEach(param => {
+      param.grantRead(this);
+      // TODO: Figure out how to determine if param is secure string for kms:Decrypt
```

Comment:
Again, if the param can know itself whether it's a secure string or not, then `grantRead()` should be able to grant that permission automatically.

Reply:
Same as my above comment. Thanks for pointing this out and grantRead here also grantsDecrypt when an encryptionKey is provided.